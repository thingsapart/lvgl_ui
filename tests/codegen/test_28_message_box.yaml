# A message box with a title, text, and some buttons.
- type: msgbox
  add_title: 'Confirmation'
  add_text: "Are you sure you want to proceed?"
  add_text: ""
  add_footer_button: "Apply"
  add_footer_button: "Cancel"
  center: []
