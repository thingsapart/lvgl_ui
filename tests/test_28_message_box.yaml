# A message box with a title, text, and some buttons.
- type: msgbox
  title: 'Confirmation'
  text: "Are you sure you want to proceed?"
  add_close_button: true
  buttons: ["Apply", "Cancel", ""]
  center: []
