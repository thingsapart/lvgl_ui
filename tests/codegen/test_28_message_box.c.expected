/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: msgbox_0 (msgbox)
    lv_obj_t* msgbox_0 = lv_msgbox_create(parent);

    lv_msgbox_add_title(msgbox_0, "Confirmation");
    lv_msgbox_add_text(msgbox_0, "Are you sure you want to proceed?");
    lv_msgbox_add_text(msgbox_0, "");
    lv_msgbox_add_footer_button(msgbox_0, "Apply");
    lv_msgbox_add_footer_button(msgbox_0, "Cancel");
    lv_obj_center(msgbox_0);

}
